tests/unit/test_federated_auth_plugin.py

```diff
@@ -172,6 +172,29 @@ def test_no_cached_token(mocker, mock_plugin_service, mock_session, mock_func, m
     assert WrapperProperties.USER.get(test_props) == _DB_USER
     assert WrapperProperties.PASSWORD.get(test_props) == _TEST_TOKEN
 
+@patch("aws_advanced_python_wrapper.federated_plugin.FederatedAuthPlugin._token_cache", _token_cache)
+def test_no_cached_token_raises_exception(mocker, mock_plugin_service, mock_session, mock_func, mock_client, mock_dialect,
```

Comment:
Didn't see any unit tests associated with the JDBC wrapper so not sure if this is appropriate.